File: src/services/facultyService.js
import apiClient from "../api/apiClient";


export const getAllFaculty = async () => {

    const response = await apiClient.get("/admin/faculty");

    return response.data;

};

export const getFacultyById = async (id) => {

    const response = await apiClient.get(`/admin/faculty/${id}`);

    return response.data;

};

export const createFaculty = async (data) => {

    const response = await apiClient.post("/faculty", data);

    return response.data;

};

export const updateFaculty = async (id, data) => {

    const response = await apiClient.put(`/faculty/${id}`, data);

    return response.data;

};

export const deleteFaculty = async (id) => {

    const response = await apiClient.delete(`/faculty/${id}`);

    return response.data;

};



export const getDashboardStats = async () => {

    const response = await apiClient.get("/faculty/dashboard");

    return response.data;

};

File: FacultyService.java
package com.campusgpt.service;

import java.util.List;

import com.campusgpt.dto.request.CreateFacultyRequest;
import com.campusgpt.dto.request.UpdateFacultyRequest;
import com.campusgpt.dto.response.FacultyResponse;

public interface FacultyService {

    void createFaculty(CreateFacultyRequest request);

    void updateFaculty(Long id, UpdateFacultyRequest request);

    void deleteFaculty(Long id);

    FacultyResponse getFacultyById(Long id);

    List<FacultyResponse> getAllFaculty();

}